lib/chef/resource/apt_repository.rb

```diff
@@ -319,38 +319,41 @@ def keyring_path
         def install_key_from_uri(key)
           key_name = key.gsub(/[^0-9A-Za-z\-]/, "_")
           keyfile_path = ::File.join(Chef::Config[:file_cache_path], key_name)
+          real_path = keyring_path
           tmp_dir = TargetIO::Dir.mktmpdir(".gpg")
           at_exit { TargetIO::FileUtils.remove_entry(tmp_dir) }
 
           if new_resource.signed_by
-            keyfile_path = keyring_path
-
             directory "/etc/apt/keyrings" do
               mode "0755"
             end
-          end
 
-          declare_resource(key_type(key), keyfile_path) do
-            source key
-            mode "0644"
-            sensitive new_resource.sensitive
-            action :create
-            verify "gpg --homedir #{tmp_dir} %{path}"
-          end
+            declare_resource(key_type(key), keyfile_path) do
```

Comment:
can we rename `def key_type` to `def key_resource_type` or something? it's so confusing to me everytime I look at this code.